http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/DataprocRuntimeMonitor.scala

```diff
@@ -137,14 +137,32 @@ class DataprocRuntimeMonitor[F[_]: Parallel](
                   // Note we don't need to check startup script results here because Dataproc
                   // won't transition the cluster to Running if a startup script failed.
 
-                  val masterInstance = instances.find(_.dataprocRole == DataprocRole.Master)
+                  val masterInstance = dataprocAndComputeInstances.find(_._1.dataprocRole == DataprocRole.Master)
 
                   masterInstance match {
-                    case Some(i) =>
-                      i.ip match {
+                    case Some((dataprocInstance, computeInstance)) =>
+                      dataprocInstance.ip match {
                         case Some(ip) =>
-                          // It takes a bit for jupyter to startup, hence wait 5 seconds before we check jupyter
-                          F.sleep(8 seconds) >> handleCheckTools(monitorContext, runtimeAndRuntimeConfig, ip, instances)
+                          for {
+                            // If the cluster is configured with worker private access, then
+                            // remove the workerPrivateAccessNetworkTag from the master node
+                            // once it is running.
+                            _ <- if (dataprocConfig.workerPrivateAccess) {
```

Comment:
Note this technically returns an `Operation` so we could poll for it.. but in practice it seems pretty quick and we are polling Jupyter immediately after this anyway.